dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/cache/AnalyticsCache.java

```diff
@@ -56,27 +51,24 @@
 {
     private static final Log log = getLog( AnalyticsCache.class );
 
-    private Cache<Grid> queryCache;
-
-    private final CacheProvider cacheProvider;
-
-    private final Environment environment;
+    private final Cache<Grid> queryCache;
 
     private final AnalyticsCacheSettings analyticsCacheSettings;
 
-    private static final int MAX_CACHE_ENTRIES = 20000;
-
-    private static final String CACHE_REGION = "analyticsResponse";
-
-    public AnalyticsCache( final CacheProvider cacheProvider, final Environment environment,
+    public AnalyticsCache( final CacheContext cacheContext,
         final AnalyticsCacheSettings analyticsCacheSettings )
     {
-        checkNotNull( cacheProvider );
-        checkNotNull( environment );
+        checkNotNull( cacheContext );
         checkNotNull( analyticsCacheSettings );
-        this.cacheProvider = cacheProvider;
-        this.environment = environment;
+
         this.analyticsCacheSettings = analyticsCacheSettings;
+        // Set a default expiration time to always expire, as the TTL will be
```

Comment:
Maybe move the cache logic to a separate method?

Reply:
That is a question for someone who knows more about `AnalyticsCacheSettings`. Here is just moved the existing logic from `init()` into the constructor. I feel in this PR it is better to not go further than that.

Reply:
Right, I think we can just keep the init method / rename it to initCache(), and call that method. Anyway, that can be done later.